crates/pixi_build_types/src/project_model.rs

```diff
@@ -0,0 +1,279 @@
+//! This module is a collection of types that represent a pixi package in a protocol
+//! format that can be sent over the wire.
+//!
+//! We need to vendor a lot of the types, and simplify them in some cases, so that
+//! we have a stable protocol that can be used to communicate in the build tasks.
+//!
+//! This is why we append a `V{version}`, to the type names, to indicate the version
+//! of the protocol.
+//!
+//! Only the Whole ProjectModel is versioned explicitly in a enum.
+//! When making a change to one of the types be sure to add another enum declaration, if it is breaking.
+use std::collections::HashMap;
+use std::path::PathBuf;
+
+use indexmap::IndexMap;
+use rattler_conda_types::{BuildNumberSpec, StringMatcher, Version, VersionSpec};
+use rattler_digest::{serde::SerializableHash, Md5, Sha256};
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+use serde_with::DisplayFromStr;
+use url::Url;
+
+/// Enum containing all versions of the project model.
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(tag = "version")]
+#[serde(rename_all = "camelCase")]
+pub enum VersionedProjectModel {
+    V1(ProjectModelV1),
+}
+
+/// The source package name of a package. Not normalized per se.
+type SourcePackageName = String;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ProjectModelV1 {
+    /// The name of the project
+    pub name: String,
+
+    /// The version of the project
+    pub version: Version,
+
+    /// An optional project description
+    pub description: Option<String>,
+
+    /// Optional authors
+    pub authors: Option<Vec<String>>,
+
+    /// The license as a valid SPDX string (e.g. MIT AND Apache-2.0)
+    pub license: Option<String>,
+
+    /// The license file (relative to the project root)
+    pub license_file: Option<PathBuf>,
+
+    /// Path to the README file of the project (relative to the project root)
+    pub readme: Option<PathBuf>,
+
+    /// URL of the project homepage
+    pub homepage: Option<Url>,
+
+    /// URL of the project source repository
+    pub repository: Option<Url>,
+
+    /// URL of the project documentation
+    pub documentation: Option<Url>,
+
+    /// Configuration for this specific project model
+    pub configuration: serde_json::Value,
+
+    pub targets: TargetsV1,
```

Comment:
Missing doc. Do we have `#![deny(docs)]` for this crate?

Reply:
Yeah we do..

Reply:
Done